tracer/src/Datadog.Trace/Configuration/ConfigurationSources/EnvironmentConfigurationSource.cs

```diff
@@ -17,6 +19,20 @@ namespace Datadog.Trace.Configuration
     /// </summary>
     public class EnvironmentConfigurationSource : StringConfigurationSource
     {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EnvironmentConfigurationSource"/> class.
+        /// </summary>
+        [PublicApi]
+        public EnvironmentConfigurationSource()
+        {
+            TelemetryFactory.Metrics.Record(PublicApiUsage.EnvironmentConfigurationSource_Ctor);
+        }
+
+        internal EnvironmentConfigurationSource(bool unused)
```

Comment:
This is sad :) Another option, clearly not saying it's a better one, could be to have an internal class inheriting from this one.

If you keep the useless parameter and if you add the parameter name when calling the ctor, could you rename the param to `uselessParamNotToUsePublicApi` or something of the sort

Reply:
Thoughts on `isInternalUse` instead of `unused`? I could see me wondering why there is an `unused` parameter in the function call, assume it is some strange legacy artifact, and try to use the call that doesn't have an `unused` parameter :) 

Or what about maybe instead of a `bool` make an `enum` to make it a bit more obvious?

Reply:
Yeah, the, internal sub-class is a better idea tbh